### PR TITLE
perf: integrate optimizations for nvidia cards

### PR DESCRIPTION
**Pull request by dignifiedquire**
based on the work from supranational

tests are not passing yet

### CODE REVIEW

**Comment by keyvank** [comment]
Hey! How faster is this?

**Comment by dignifiedquire** [comment]
> Hey! How faster is this?

I don't know yet, need to figure out how to avoid opencl taking > 20min to compile with these optimizations 😭

**Comment by vmx** [comment]
This is ready for review. The review is easier if the two commits are reviewed individually.

File: ec-gpu-gen/src/lib.rs [hunk 1]
```diff
@@ -219,6 +206,10 @@ where
     .replace("FIELD", name)
 }
 
+pub fn common() -> String {
```
Comment (dignifiedquire) [inline ec-gpu-gen/src/lib.rs:209]: cany you add some docs about this please?

Comment (vmx) [reply ec-gpu-gen/src/lib.rs:214]: Done.

**Review by dignifiedquire — COMMENTED** [review]
lgtm (can't approve my own PR)

**Comment by vmx** [comment]
@dignifiedquire please see the minimal changes since your last review. Now [bellperson successfully runs all tests on CI](https://app.circleci.com/pipelines/github/filecoin-project/bellperson/1373/workflows/a95b5157-bb5a-43ff-b044-91ea8203da8c) with this version (OpenCL and CUDA).

If you could please give another thumbs up, I'll fixup the last commits and merge :)

**Pull request merged**